Lib/test/test_tstring.py

```diff
@@ -0,0 +1,441 @@
+import ast
+import unittest
+
+from string.templatelib import Template, Interpolation
+
+
+def convert(value, conversion):
+    if conversion == "a":
+        return ascii(value)
+    elif conversion == "r":
+        return repr(value)
+    elif conversion == "s":
+        return str(value)
+    return value
+
+
+def f(template):
+    parts = []
+    for item in template:
+        match item:
+            case str() as s:
+                parts.append(s)
+            case Interpolation(value, _, conversion, format_spec):
+                value = convert(value, conversion)
+                value = format(value, format_spec)
+                parts.append(value)
+    return "".join(parts)
+
+
+class TestTString(unittest.TestCase):
+    def assertAllRaise(self, exception_type, regex, error_strings):
```

Comment:
Optional: this is like the one in `test_fstring.py`, we could move it to `test.support` and it would be useful for other tests too.

Fine to ignore though to keep this PR a bit smaller. Or we can do in a separate PR first so the `test_fstring.py` change can be backported.